addons/udes_stock/models/stock_picking.py

```diff
@@ -963,19 +963,17 @@ def update_picking(
             )
 
     def _requires_backorder(self, mls):
-        """Checks if a backorder is required
-        by checking if all move.lines
-        within a picking is present in mls
+        """
+        Checks if a backorder is required by checking if all moves within a picking are present
+        in the movelines. Cancelled moves are not considered.
         """
         mls_moves = mls.mapped("move_id")
-        # return (mls_moves | self.move_lines) != mls_moves or \
-        # (mls | self.move_line_ids) != mls or \
-        # self.mapped('move_lines.move_orig_ids').filtered(lambda x: x.state not in ('done', 'cancel'))
-        for move in self.move_lines:
+        
+        for move in self.move_lines.filtered(lambda mv: mv.state != "cancel"):
```

Comment:
Q: Do we also want to move the mls into a backorder if the remaining moves are done? Not sure just a question.

Reply:
I think this would be covered in the check below, I think for what we need in this story just filtering out cancelled moves should be enough anyway.

Obviously there is a lot we can do better here in general.